stdlib/public/core/StringCreate.swift

```diff
@@ -50,18 +50,51 @@ internal func _allASCII(_ input: UnsafeBufferPointer<UInt8>) -> Bool {
 }
 
 extension String {
-  @usableFromInline
-  internal static func _fromASCII(
-    _ input: UnsafeBufferPointer<UInt8>
-  ) -> String {
+  
+  internal static func _uncheckedFromASCII(_ input: UnsafeBufferPointer<UInt8>)
+  -> String {
     _internalInvariant(_allASCII(input), "not actually ASCII")
 
     if let smol = _SmallString(input) {
       return String(_StringGuts(smol))
     }
-
+    
+    return __StringStorage.create(
+      initializingFrom: input, isASCII: true
+    ).asString
+  }
+  
+  internal static func _fromASCIIRepairing(
+    _ input: UnsafeBufferPointer<UInt8>
+  ) -> (result: String, repairsMade: Bool) {
+    
+    if let smol = _SmallString(input) {
+      return (String(_StringGuts(smol)), false)
```

Comment:
Where does the error correction happen here?

It seems like we need some `String(decoding:x as:ASCII.self)` test cases for invalid small and large strings.